### PR TITLE
Issue #3400 - node policy change

### LINKED ISSUE #3400: Already deployed containers should not be stopped when node property is changed
### Is your feature request related to a problem? Please describe.

User is finding that when a new property is added in the node policy to deploy an additional service container, all the previously running agreements are cancelled, that result into already running containers getting stopped. New agreements are formed and they start all over again. 

### Describe the solution you'd like.

Expectation is that already running containers should not be stopped and only newly added container should be started.

### Describe alternatives you've considered

_No response_

### Additional context.

This is important when user selectively wants to deploy additional application later on by adding/removing the node policy properties.

### PR DESCRIPTION
**Pull request by johnwalicki**
Signed-off-by: John Walicki <johnwalicki@gmail.com>

# Pull Request Template

## Description

Fixes #3400

Call governAgreements() - let it determine if the updated policy is compatible with established agreements

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have checked my code and corrected any misspellings
- [x] I have tagged the reviewers in a comment below incase my pull request is ready for a review
- [x] I have signed the commit message to agree to Developer Certificate of Origin (DCO) (to certify that you wrote or otherwise have the right to submit your contribution to the project.) by adding "--signoff" to my git commit command.

### CODE REVIEW

File: governance/node.go [hunk 1]
```diff
@@ -80,49 +79,9 @@ func (w *GovernanceWorker) handleNodeUserInputUpdated(svcSpecs persistence.Servi
...
-				// send the event to the container in case it has started the workloads.
-				w.Messages() <- events.NewGovernanceWorkloadCancelationMessage(events.AGREEMENT_ENDED, events.AG_TERMINATED, ag.AgreementProtocol, agreementId, ag.GetDeploymentConfig())
-				// clean up microservice instances if needed
-				w.handleMicroserviceInstForAgEnded(agreementId, false)
-			}
-		}
-	}
+	// Make sure that every agreement we have is in a valid state or proceeding to valid states in a timely fashion. If not,
+	// cancel the agreement and allow the agbots to re-make them if necessary.
+	w.governAgreements()
```
Comment (linggao) [inline governance/node.go:84]: governAgreements check the compatibility of of the policies. It does not handle the user input changes.

Comment (johnwalicki) [reply governance/node.go:84]: Tracing the logs (generated by an IEAM 4.3 agent v2.29.0), the logic flows and error messages point to this function and causes the agreements to be cancelled.

(I see a PR from 13 months ago that changed the behavior of all of this.  Is the 2.29 branch that much older than 2.30 ?)  It's possible I was tracing the logs generated from a 2.29.0 agent and the 2.30.0 code base is substantially divergent. 

https://github.com/open-horizon/anax/commit/2b0219fbfac1ef378c86b75c4750860a68e539cf#diff-8f7cb8d1fb0d5b06387d5655c337658bd5ff41ef9317f1499442b7683b55a693L47

Comment (linggao) [reply governance/node.go:84]: One way to test it is:
1. start e2edev test.
2. hzn userinput list > /tmp/ui.json
3.  change a value in /tmp/ui.json
4. hzn userinput add -f /tmp/ui.json  see if the agreement got canceled or not. The agreements that uses the value are supposed to be canceled.

**Comment by LiilyZhang** [comment]
This code change is not needed for issue 3400

**Pull request closed without merging**